### DISCUSSION TITLE
High frame rate VideoWriter issue

### DISCUSSION THREAD

**Discussion by marcelo-moglie** [bonsai-rx/bonsai#1669] [Q&A] [answered]
Hi,

I am using a Basler camera to acquire 1100x 800 px video  at 300 fps. Everything works fine when I acquire at 100 fps but the computer crashes after 20 min acquiring at 300 fps. I have checked the resource monitor and it seems that memory usage keeps increasing until it runs out at some point (192 GB RAM installed), while the SSD remains at its lowest usage.  I set the VideoWriter node to be buffered but I believe it is buffering the whole session and only starts copying to the SSD once I stop recording. I have installed FFMPEG but it doesn't seem to get any better. Any help will be much appreciated.

All the best,
Marcelo

**Comment by bruno-f-cruz** [comment]
You computer doesn't seem to be able to keep up with the encoding (which, given the fps and resolution doesn't surprise me). FFMPEG might be the solution but you might have to play with the codec settings to get something that works. What settings are you currently using?

**Comment by marcelo-moglie** [comment]
Hi Bruno,

Thanks a lot for your quick response. I am just using the videowriter node setting the FourCC to FMP4. What would be the best way to ensure that the codec is working in the intended way?
I was wondering if the video should be gradually stored in the SSD as it is generated and whether I could change the buffering size in the videowriter node. The video file is created when the recording starts but its size stays 0 kB until I finish the recording.

Thanks,
Marcelo



________________________________

**Reply by marcelo-moglie** [reply, marked as answer]
Hi Bruno, 
I have managed to have it running without problems by making use of the FFmpeg file in the Bonsai Gallery and playing with the codecs. I found the preset options for H.264 encoding very straightforward to use. Thanks!
